src/main/java/eu/dissco/orchestration/backend/controller/SourceSystemController.java

```diff
@@ -58,9 +58,13 @@ public ResponseEntity<JsonApiWrapper> updateSourceSystem(Authentication authenti
     var sourceSystem = getSourceSystemFromRequest(requestBody);
     var id = prefix + '/' + suffix;
     log.info("Received update request for source system: {}", id);
-    String path = SANDBOX_URI + servletRequest.getRequestURI();
+    String path = appProperties.getBaseUrl() + servletRequest.getRequestURI();
     var result = service.updateSourceSystem(id, sourceSystem, path);
-    return ResponseEntity.ok(result);
+    if (result == null) {
+      return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+    } else {
+      return ResponseEntity.ok(result);
```

Comment:
can take this out of the else block too

Reply:
hmm I like it in an `else` block. It makes the distinction clear, plus if we ever change anything it won't automatically become the `.ok(result)`.